### PR TITLE
added Refined Smeltery

### PR DESCRIPTION
**Pull request by J3fftw1**
Changed Recipe from Refined Iron 
Might change the Multiblock later(Dont really like the design)

### CODE REVIEW

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            }
+        }
+
+        outputInv.addItem(output);
+        p.getWorld().playSound(p.getLocation(), Sound.BLOCK_LAVA_POP, 1, 1);
+        p.getWorld().playEffect(b.getLocation(), Effect.MOBSPAWNER_FLAMES, 1);
+
+        Block fire = b.getRelative(BlockFace.DOWN).getRelative(BlockFace.DOWN);
+        fire.getWorld().playEffect(fire.getLocation(), Effect.STEP_SOUND, fire.getType());
+        fire.setType(Material.AIR);
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java:96]: Ouch, fire is removed after every craft

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            items.add(recipes.get(i)[0]);
+            items.add(recipes.get(i + 1)[0]);
+        }
+
+        return items;
+    }
+
+    @Override
+    public void onInteract(Player p, Block b) {
+        Block dispBlock = b.getRelative(BlockFace.DOWN);
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java:58]: Check if b is the stone brick wall

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java (hunk 1 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java:96]: Imagine LX being easy
they kept complaining change recipe now get fucked by placing fire every time

**Review by WalshyDev — CHANGES_REQUESTED** [review]
Has this been tested?

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java (hunk 1 above)
Comment (WalshyDev) [reply src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java:96]: LULW love it

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java (hunk 2 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java:58]: it only works with stone brick wall
tested it just now

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java (hunk 1 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java:96]: you know what would be mean if we check if there is another dispenser next to the fire if so block smelting
so we can block automation

**Comment by WalshyDev** [comment]
Has this been tested properly? Throws message on invalid recipe, works with correct recipe, extinguishes fire, requires the fire actually be there, etc?

My main thing is, does it actually require the fire there

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java (hunk 1 above)
Comment (WalshyDev) [reply src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java:96]: Oh Jesus

**Comment by J3fftw1** [comment]
> 
> 
> Has this been tested properly? Throws message on invalid recipe, works with correct recipe, extinguishes fire, requires the fire actually be there, etc?
> 
> My main thing is, does it actually require the fire there

it actually needs fire it doesnt throw any messages though, will add them rn

**Comment by WalshyDev** [comment]
ty

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        for (int i = 0; i < inputs.size(); i++) {
+            if (canCraft(inv, inputs, i)) {
+                ItemStack output = RecipeType.getRecipeOutputList(this, inputs.get(i)).clone();
+
+                if (Slimefun.hasUnlocked(p, output, true)) {
+                    Inventory outputInv = findOutputInventory(output, dispBlock, inv);
+
+                    if (outputInv != null) {
+                        craft(p, b, inv, inputs.get(i), output, outputInv);
+                    } else SlimefunPlugin.getLocalization().sendMessage(p, "machines.full-inventory", true);
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java]: Don't put crap on the same line as an else

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            }
+        }
+
+        SlimefunPlugin.getLocalization().sendMessage(p, "machines.unknown-material", true);
+    }
+
+    private void craft(Player p, Block b, Inventory inv, ItemStack[] recipe, ItemStack output, Inventory outputInv) {
+        for (ItemStack removing : recipe) {
+            if (removing != null) {
+                InvUtils.removeItem(inv, removing.getAmount(), true, stack -> SlimefunUtils.isItemSimilar(stack,
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java]: should just newline the lambda arg

File: src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        Block fire = b.getRelative(BlockFace.DOWN).getRelative(BlockFace.DOWN);
+        fire.getWorld().playEffect(fire.getLocation(), Effect.STEP_SOUND, fire.getType());
+        fire.setType(Material.AIR);
+    }
+
+    private boolean canCraft(Inventory inv, List<ItemStack[]> inputs, int i) {
+        for (ItemStack converting : inputs.get(i)) {
+            if (converting != null) {
+                for (int j = 0; j < inv.getContents().length; j++) {
+                    if (j == (inv.getContents().length - 1) && !SlimefunUtils.isItemSimilar(converting,
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/machine/RefinedSmeltery.java]: \n&& ...

**Review by WalshyDev — CHANGES_REQUESTED** [review]

**Pull request merged**